pkg/storer/internal/reserve/reserve.go

```diff
@@ -100,9 +101,12 @@ func New(
 //     if the new chunk has a higher stamp timestamp (regardless of batch type and chunk type, eg CAC & SOC).
 func (r *Reserve) Put(ctx context.Context, chunk swarm.Chunk) error {
 
```

Comment:
There are too many changes and it's seems that we are no longer handling cases we used to before. It also feels like we've changed the tests until they started passing. What I would like to see is we revert all these changes. 

Maybe we can handle this by, in the case of a same chunk address collision, we first check by chunk type. If it's a soc then we compare the identity addresses by computing them. If some checks pass, we store the new chunk and return early.

Reply:
Ping me for help on deep diving these changes.

Reply:
there was some misunderstanding by what sort of logic the chunks should be stored in the reserve.
It was checked by batch ID when only the valid stamps matter. if there is a valid stamp which was not stored before you must store it.